jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/BusinessRuleAssignmentTest.java

```diff
@@ -195,6 +197,59 @@ public void testAssignmentsWhenNoRuleWasFired() throws Exception {
         User actualOwner = task.getTaskData().getActualOwner();
         Assertions.assertThat(actualOwner).isNull();
     }
+    
+    @Test
+    public void testAssignmentAssignToBobbaFetBasedOnDataInput() {
+        String str = "(with (new Task()) { priority = 55, taskData = (with( new TaskData()) { } ), ";
+        str += "peopleAssignments = (with ( new PeopleAssignments() ) { potentialOwners = [new User('Darth Vader'), new User('Bobba Fet'), new User('Luke Cage')  ],businessAdministrators = [ new User('Administrator') ],}),";
+        str += "name =  'Bobbas dedicated tasks' })";
+        
+        Map<String, Object> input = new HashMap<>();
+        input.put("name", "Bobbas dedicated tasks");
+        
+        createAndAssertTask(str, "Bobba Fet", 3, input, "Bobba Fet", "Darth Vader", "Luke Cage"); 
+        
+        input.clear();
+        // another task
+        createAndAssertTask(str, null, 3, input, "Bobba Fet", "Darth Vader", "Luke Cage"); 
+         
+    }
+    
+    @Test
+    public void testAssignmentAssignToLukeCageBasedOnDataInputAndDeadlines() {
```

Comment:
Please add timeout to this test method. For some reason the EntityManager was closed when the `ExecuteDeadlinesCommand` was being executed:

```
WARN  Could not commit session
12:26:14 java.lang.IllegalStateException: EntityManagerFactory is closed
12:26:14 	at org.hibernate.internal.SessionFactoryImpl.validateNotClosed(SessionFactoryImpl.java:531)
12:26:14 	at org.hibernate.internal.SessionFactoryImpl.createEntityManager(SessionFactoryImpl.java:613)
12:26:14 	at org.hibernate.internal.SessionFactoryImpl.createEntityManager(SessionFactoryImpl.java:154)
12:26:14 	at org.drools.persistence.jpa.AbstractPersistenceContextManager.getCommandScopedEntityManager(AbstractPersistenceContextManager.java:99)
12:26:14 	at org.jbpm.services.task.persistence.JPATaskPersistenceContextManager.beginCommandScopedEntityManager(JPATaskPersistenceContextManager.java:49)
12:26:14 	at org.jbpm.services.task.persistence.TaskTransactionInterceptor.execute(TaskTransactionInterceptor.java:73)
12:26:14 	at org.jbpm.services.task.persistence.TaskTransactionInterceptor.execute(TaskTransactionInterceptor.java:49)
12:26:14 	at org.drools.core.command.impl.AbstractInterceptor.executeNext(AbstractInterceptor.java:39)
12:26:14 	at org.drools.persistence.jta.TransactionLockInterceptor.execute(TransactionLockInterceptor.java:73)
12:26:14 	at org.drools.persistence.jta.TransactionLockInterceptor.execute(TransactionLockInterceptor.java:45)
12:26:14 	at org.drools.core.command.impl.AbstractInterceptor.executeNext(AbstractInterceptor.java:39)
12:26:14 	at org.drools.persistence.jpa.OptimisticLockRetryInterceptor.internalExecute(OptimisticLockRetryInterceptor.java:102)
12:26:14 	at org.drools.persistence.jpa.OptimisticLockRetryInterceptor.execute(OptimisticLockRetryInterceptor.java:83)
12:26:14 	at org.drools.persistence.jpa.OptimisticLockRetryInterceptor.execute(OptimisticLockRetryInterceptor.java:44)
12:26:14 	at org.drools.core.runtime.InternalLocalRunner.execute(InternalLocalRunner.java:37)
12:26:14 	at org.drools.core.runtime.InternalLocalRunner.execute(InternalLocalRunner.java:41)
12:26:14 	at org.jbpm.services.task.commands.TaskCommandExecutorImpl.execute(TaskCommandExecutorImpl.java:49)
12:26:14 	at org.jbpm.services.task.impl.TaskDeadlinesServiceImpl$ScheduledTaskDeadline.call(TaskDeadlinesServiceImpl.java:284)
12:26:14 	at org.jbpm.services.task.impl.TaskDeadlinesServiceImpl$ScheduledTaskDeadline.call(TaskDeadlinesServiceImpl.java:229)
12:26:14 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
12:26:14 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
12:26:14 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
12:26:14 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
12:26:14 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
12:26:14 	at java.lang.Thread.run(Thread.java:748)
```

And then the test timeouted.

But the real issue was that no rules were applied for the newly added test scenario:

```
[main|org.jbpm.services.task.assignment.impl.AssignmentServiceImpl] WARN  Strategy BusinessRuleAssignmentStrategy [releaseId=org.jbpm.task:assignment-rules:1.0] did not return any assignment for task TaskImpl [id=0, name=Simple Test Task]
```

Probably the issue is with the potentialOwners part in the rule itself. Luke Cage is not assigned in potentialOwners of the task, but in the potentialOwners section of the deadline. Then the rule won't fire I guess.

```
PeopleAssignments(potentialOwners contains new UserImpl('Luke Cage')) from $task.getPeopleAssignments()
```

Not sure about the EntityManagerFactory closed issue though.

Reply:
@MarianMacik yes, I've fixed that issue (missing TaskDeadLines reset invocation in the clean method), but it's still on @mswiderski  repository for him to merge.
https://github.com/mswiderski/jbpm/pull/9/files

Reply:
Why do you need to reset it? I cannot see the connection between resetting it and the EntityManagerFactory being closed.

Reply:
It's something weird. This issue was only present when a another test was executed before this new one in the test suite. PersistenceContext and Scheduler are initialized (due to getTaskService call in the setup method) but not reset. 
So _TaskDeadlinesServiceImpl.instance_ has to be set to _null_ after each test; if not, this line https://github.com/kiegroup/jbpm/blob/15932b610e844de410d14c8f5b44be3ca815977a/jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/HumanTaskConfigurator.java#L165-L167

returns not null, and it's not initialize, using an instance (TaskCommandExecutorImpl) that it's already closed.

Reply:
Ah, ok. Apparently then the CommandExecutor which remains in the TaskDeadlinesServiceImpl contains the old environment and not the new one. Thanks for the clarification.